### PR TITLE
feat: expose new props to add custom color in Chip component

### PR DESCRIPTION
**Pull request by lukewalczak**
Summary: This PR will add two new props to customize color of Chip component

screenshot 2019-01-14 at 16 04 49

### Motivation

Issue no #725

### CODE REVIEW

File: src/components/Chip.js [hunk 1]
```diff
@@ -175,6 +180,12 @@ class Chip extends React.Component<Props, State> {
       .rgb()
       .string();
 
+    const underlayColor =
+      color(selectedColor)
```
Comment (satya164) [inline src/components/Chip.js]: Won't this crash if `selectedColor` is not passed?

File: src/components/Chip.js [hunk 2]
```diff
@@ -151,20 +156,20 @@ class Chip extends React.Component<Props, State> {
 
     const borderColor =
       mode === 'outlined'
-        ? color(dark ? white : black)
+        ? color(selectedColor || color(dark ? white : black))
```
Comment (satya164) [inline src/components/Chip.js]: seems there is double wrapping of color. change to:

```suggestion
        ? color(selectedColor !== undefined ? selectedColor : dark ? white : black)
```

File: src/components/Chip.js [hunk 3]
```diff
@@ -151,20 +156,20 @@ class Chip extends React.Component<Props, State> {
...
             .alpha(0.29)
             .rgb()
             .string()
         : backgroundColor;
     const textColor = disabled
       ? colors.disabled
-      : color(colors.text)
+      : (selectedColor ? color(selectedColor) : color(colors.text))
```
Comment (satya164) [inline src/components/Chip.js]: This reads better

```suggestion
      : color(selectedColor !== undefined ? selectedColor : colors.text)
```

File: src/components/Chip.js [hunk 4]
```diff
@@ -151,20 +156,20 @@ class Chip extends React.Component<Props, State> {
...
           .alpha(0.87)
           .rgb()
           .string();
     const iconColor = disabled
       ? colors.disabled
-      : color(colors.text)
+      : (selectedColor ? color(selectedColor) : color(colors.text))
```
Comment (satya164) [inline src/components/Chip.js]: This reads better
```suggestion
      : color(selectedColor !== undefined ? selectedColor : colors.text)
```

**Comment by ferrannp** [comment]
Can someone else take a look at this?

**Comment by lukewalczak** [comment]
Working on that

**Pull request merged**